pkg/portal/graphql/app.go

```diff
@@ -147,11 +148,18 @@ var nodeApp = node(
 	&model.App{},
 	func(ctx context.Context, id string) (interface{}, error) {
 		gqlCtx := GQLContext(ctx)
+		// check for existence of app
```

Comment:
I don't think this change is needed? The data loader should translate the errors to null data, so client-side only need to check whether the node is null.

Reply:
Non-existing app will trigger `Forbidden` error, so this is used to distinguish `not exist` from `no access`

Reply:
It also prevent the error in first screenshot

Reply:
We should be allow client-side to distinguish between a non-existant app and an inaccessible app to prevent information leakage. Instead, treat both cases as not found.

For the pg error, I think we can use `utf8.ValidString` to validate all decode IDs.

Reply:
`utf8.ValidString` cannot block `0x00` (when cast string seems like `0x00` is omitted), I changed to throw forbidden if app is not on list

Reply:
We should keep simply returning null for non-existant app and inaccessible app.

Reply:
> simply returning null for non-existant app and inaccessible app

no error for inaccessible app?

Reply:
Yes, there should be no errors returned from API for inaccessible/not found apps.